Wilab-io/sample-apps
# [Login] Develop the login screen and logic

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

## Description

I dockerized the application and included a Postgres 15 container in the docker-compose package.
I also implemented the connection to that DB using env variables. (the default values should be fine though)
Then I created some initialization scripts for the DB to create the tables and their relationships.

I then created a script in the code that connects to the DB and makes the admin user, it is written in the code because we need to hash the password before storing it in the DB.

In the frontend, I developed the login screen and its functionality, if the user or password is wrong, a message will appear on top of the Login button. If everything is OK, you'll be redirected to the home page.

If you try to go to a page and aren't logged in, you'll be redirected to `/login`.
Also, if you try to go to `/login` when logged in, you'll be immediately redirected to the home page.

## How to test

Ask me for the .env file with my credentials.
Then go to the project and run

```bash
docker compose up --build
```
(This will take up to 10 minutes because it uses CPU inside of Docker)

When you see the following log, you are ready to open the page.

`INFO:     Uvicorn running on http://0.0.0.0:7860 (Press CTRL+C to quit)`
